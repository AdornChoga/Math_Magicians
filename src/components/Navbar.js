import '../stylesheets/Navbar.css';

const Navbar = () => {
  const navLinks = [
    {
      id: 1,
      path: '',
      textContent: 'Home',
    },
    {
      id: 2,
      path: '',
      textContent: 'Calculator',
    },
    {
      id: 3,
      path: '',
      textContent: 'Quote',
    },
  ];
  return (
    <nav className="nav-bar">
      <h1>Math Magicians</h1>
      <ul>
        {navLinks.map((link) => (
          <li key={link.id}>{link.textContent}</li>
        ))}
      </ul>
    </nav>
  );
};

export default Navbar;
