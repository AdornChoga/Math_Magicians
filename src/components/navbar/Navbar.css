* {
  box-sizing: border-box;
}

html {
  font-size: 62.5%;
}

body {
  padding: 0;
  margin: 0;
}

.nav-bar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  position: fixed;
  top: 0;
  left: 0;
  background-color: rgb(124, 65, 65);
  padding: 0.8em 2em;
}

.nav-bar h1 {
  margin: 0;
  font-size: 2.8rem;
}

.nav-bar ul {
  display: flex;
  list-style-type: none;
  gap: 1rem;
  margin: 0;
}

.nav-bar ul li {
  font-size: 1.8rem;
  font-weight: bold;
  cursor: pointer;
}

.nav-bar ul li + li {
  border-left: 0.2rem solid rgb(245, 242, 242);
  padding-left: 1rem;
}

a {
  text-decoration: none;
  color: rgb(13, 13, 82);
}

a:hover {
  text-decoration: underline white;
}
