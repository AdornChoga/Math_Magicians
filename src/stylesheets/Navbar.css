* {
  box-sizing: border-box;
}

html {
  font-size: 62.5%;
}

body {
  padding: 0;
  margin: 0;
}

.nav-bar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  position: fixed;
  top: 0;
  left: 0;
  background-color: rgb(226, 159, 120);
  padding: 0.8em 2em;
}

.nav-bar h1 {
  margin: 0;
  font-size: 2.8rem;
}

.nav-bar ul {
  display: flex;
  list-style-type: none;
  gap: 1rem;
  margin: 0;
}

.nav-bar ul li {
  font-size: 1.8rem;
  font-weight: bold;
  cursor: pointer;
}

.nav-bar ul li + li {
  border-left: 0.2rem solid rgb(109, 107, 107);
  padding-left: 1rem;
}

.nav-bar ul li:hover {
  text-decoration: underline white;
}
