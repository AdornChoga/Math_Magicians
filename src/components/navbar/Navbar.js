import './Navbar.css';
import { NavLink } from 'react-router-dom';

const Navbar = () => {
  const navLinks = [
    {
      id: 1,
      path: '/',
      textContent: 'Home',
    },
    {
      id: 2,
      path: 'calculator',
      textContent: 'Calculator',
    },
    {
      id: 3,
      path: 'quote',
      textContent: 'Quote',
    },
  ];
  return (
    <nav className="nav-bar">
      <h1>Math Magicians</h1>
      <ul>
        {navLinks.map((link) => (
          <li key={link.id}>
            <NavLink to={link.path}>
              {link.textContent}
            </NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Navbar;
