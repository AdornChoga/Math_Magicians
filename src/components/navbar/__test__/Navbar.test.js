import { render, screen, cleanup } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from '../Navbar';

afterEach(cleanup);

describe('Tests for nav bar rendering and snapshot matching.', () => {
  test('Is navbar rendered?', () => {
    render(<Router><Navbar /></Router>);
    const navElement = screen.getByTestId('nav');
    expect(navElement).toBeInTheDocument();
  });

  test('Is navbar snapshot created/matched?', () => {
    const tree = renderer.create(<Router><Navbar /></Router>);
    expect(tree).toMatchSnapshot();
  });
});
