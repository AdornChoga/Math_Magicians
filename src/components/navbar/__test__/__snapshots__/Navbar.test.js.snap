// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for nav bar rendering and snapshot matching. Is navbar snapshot created/matched? 1`] = `
<nav
  className="nav-bar"
  data-testid="nav"
>
  <h1>
    Math Magicians
  </h1>
  <ul>
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className=""
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className=""
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
